PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.core/src/com/microsoft/azuretools/core/survey/SurveyNotificationPopup.java

```diff
@@ -0,0 +1,127 @@
+/**
+ * Copyright (c) Microsoft Corporation
+ *
+ * All rights reserved.
+ *
+ * MIT License
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files
+ * (the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge,
+ * publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR
+ * ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH
+ * THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+package com.microsoft.azuretools.core.survey;
+
+import org.eclipse.mylyn.commons.ui.dialogs.AbstractNotificationPopup;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.events.SelectionAdapter;
+import org.eclipse.swt.events.SelectionEvent;
+import org.eclipse.swt.graphics.Font;
+import org.eclipse.swt.graphics.FontData;
+import org.eclipse.swt.graphics.Image;
+import org.eclipse.swt.layout.GridData;
+import org.eclipse.swt.layout.GridLayout;
+import org.eclipse.swt.widgets.Button;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Display;
+import org.eclipse.swt.widgets.Label;
+
+import com.microsoft.azuretools.core.Activator;
+
+public class SurveyNotificationPopup extends AbstractNotificationPopup {
+	private static final String SURVEY_WINDOW_TITLE = "Azure Toolkit Survey";
+	private static final String SURVEY_TEXT_HEADER = "Enjoy Azure Toolkits?";
+	private static final String SURVEY_TEXT_BODY = "Your feedback is important, take a minute to fill out our survey.";
+	private static final String PUT_OFF_BUTTON_TEXT = "Not now";
+	private static final String TAKE_SURVEY_BUTTON_TEXT = "Give feedback";
+
+	private Runnable takeSurveyAction;
+	private Runnable putOffAction;
+	private boolean isSurveyButtonClicked = false;
+
+	public SurveyNotificationPopup(Display display, Runnable takeSurveyAction, Runnable putOffAction) {
```

Comment:
Please provide another button `neverShowAgain` to allow user disable pop-up survey